hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -216,7 +216,15 @@ public ClientConnection getConnection(Address target) {
         if (target == null) {
             return null;
         }
-        return connections.get(target);
+        ClientConnection connection = connections.get(target);
+        if (connection != null && !connection.isAlive()) {
```

Comment:
for a connection to be not alive, connection must have been closed, and it's only closed via destroyConnection(). In which sequence of events can this case happen?